website/docs/tutorials/08-interacting-with-starknet/01-invoke.md

```diff
@@ -0,0 +1,63 @@
+## Details
+This cheatcode allows you to send an invoke transaction with `@external` function entrypoint.
+
+### Signing
+Credentials for paying the fee are needed, which are: 
+- Account address (which pays the fee)
+- Private key for that account (from ArgentX, Braavos etc.) - in hex string (i.e. 0xa1c359ef) form. This can be provided with `PROTOSTAR_ACCOUNT_PRIVATE_KEY` env variable or with a file on local filesystem containing that key in plaintext, in which case you should use `--private-key-path`.
+
+Custom signing logic is made possible by using custom signers - see details [here](https://docs.swmansion.com/protostar/docs/tutorials/deploying/cli#2-using-a-custom-signer-class).
+
+
+
+:::warning
+Setting `max-fee` to `auto` is discouraged, since it may incur extra unexpected costs.
+:::
+
+## Usage
+```
+protostar invoke [-h] [--account-address ACCOUNT_ADDRESS] [--private-key-path PRIVATE_KEY_PATH] [--signer-class SIGNER_CLASS] [--gateway-url GATEWAY_URL] [--chain-id CHAIN_ID] [--network NETWORK] [--contract-address CONTRACT_ADDRESS] [--function FUNCTION] [--inputs [INPUTS ...]]
+                                   [--max-fee MAX_FEE] [--wait-for-acceptance]
+
+Arguments:
+  -h, --help            show this help message and exit
+  --account-address ACCOUNT_ADDRESS
+                        Account address
+  --private-key-path PRIVATE_KEY_PATH
+                        Path to the file, which stores your private key (in hex representation) for the account.
+                        Can be used instead of PROTOSTAR_ACCOUNT_PRIVATE_KEY env variable.
+  --signer-class SIGNER_CLASS
+                        Custom signer class module path.
+  --gateway-url GATEWAY_URL
+                        The URL of a StarkNet gateway. It is required unless `--network` is provided.
+  --chain-id CHAIN_ID   The chain id. It is required unless `--network` is provided.
+  --network NETWORK, -n NETWORK
+                        The name of the StarkNet network.
+                        It is required unless `--gateway-url` is provided.
+
+                        Supported StarkNet networks:
+                        - `testnet`
+                        - `mainnet`
+                        - `alpha-goerli`
+                        - `alpha-mainnet`
+  --contract-address CONTRACT_ADDRESS
+                        The address of the contract being called.
+  --function FUNCTION   The name of the function being called.
+  --inputs [INPUTS ...]
+                        Inputs to the function being called, represented by a list of space-delimited values.
+  --max-fee MAX_FEE     The maximum fee that the sender is willing to pay for the transaction. Provide "auto" to auto estimate the fee.
+  --wait-for-acceptance
+                        Waits for transaction to be accepted on chain.
```

Comment:
We need to automate it somehow. It's going to be hard to maintain, and we will quickly introduce inconsistencies between docs and the CLI. It would be nice to have a script that generates this content (maybe in the form of table) and that markdown that could be imported to this doc.

https://docusaurus.io/docs/markdown-features/react#importing-markdown

Reply:
Every bit of automation is nice, we should throw that in the next milestone. That's out of scope of this PR though - should i remove this bit of reference here?

Reply:
I'd replace it with a link to the CLI reference for the time being, e.g.
https://docs.swmansion.com/protostar/docs/cli-reference#invoke

Reply:
What about example usage? This can easily get outdated as well

Reply:
IMHO example is fine, but I would resist copypasting help info into our docs. I think it'd be better to describe with prose what are the crucial parameters and showcase on examples what you can do with this.

Reply:
Examples are a bit more stable. We can think about introducing example testing step to our CI/CD pipeline. CLI reference is less stable because it contains descriptions, custom types etc.

Reply:
left out the reference (added a link) and kept the example